backend/uclapi/workspaces/tests_archive.py

```diff
@@ -0,0 +1,270 @@
+
+    def run_update(self, data, reset=True):
+        self.archive = OccupEyeArchive(endpoint=TestEndpoint(data))
+        if reset:
+            self.archive.reset()
+        self.archive.update()
+
+    @freeze_time("2020-01-02")
+    def test_sensor_replacement(self):
+        with open(os.path.join(__location__, "tests_archive_edge_cache.json"), encoding="utf-8") as f:
+            self.run_update(json.load(f))
+        sensor_replacement = SensorReplacements.objects.all().values()
+        self.assertEqual(len(sensor_replacement), 1)
+        sensor_replacement = sensor_replacement[0]
+
+        del sensor_replacement["id"]
+        self.assertDictEqual(sensor_replacement, {"datetime": datetime(2020, 1, 2, 0, 0),
+                                                  "hardware_id": 109006,
+                                                  "survey_device_id": 1039,
+                                                  "survey_id": 10,
+                                                  "sensor_id": 10629006})
+
+    def test_survey_change(self):
+        self.maxDiff = None
+        with freeze_time("2020-01-02"):
+            with open(os.path.join(__location__, "tests_archive_edge_cache.json"), encoding="utf-8") as f:
+                self.run_update(json.load(f))
+        with freeze_time("2020-01-04"):
+            with open(os.path.join(__location__, "tests_archive_edge_survey_cache.json"), encoding="utf-8") as f:
+                self.run_update(json.load(f), reset=False)
+
+        survey_change = SurveyChanges.objects.all().order_by("datetime").values()
+        self.assertEqual(len(survey_change), 2)
+        sensor_replacement = survey_change[1]
+
+        del sensor_replacement["id"]
+        del sensor_replacement["datetime"]
+        self.assertDictEqual(sensor_replacement, {"active": True,
+                                                  "end_datetime": datetime(2050, 1, 1, 15, 0),
+                                                  "name": "Replacement Room 2",
+                                                  "start_datetime": datetime(2010, 1, 1, 9, 0),
+                                                  "survey_id": 10})
+
+    def tearDown(self):
+        Surveys.objects.all().delete()
+        Sensors.objects.all().delete()
+        Historical.objects.all().delete()
+        SensorReplacements.objects.all().delete()
+        SurveyChanges.objects.all().delete()
+
+
+@freeze_time("2020-02-01")
+class OccupEyeArchiveViewsTestCase(TestCase):
+
+    @patch("workspaces.occupeye.archive.FIRST_OCCUPEYE_INSTALLATION",
+           datetime.strptime("2020-01-01T00:00:00+0000", "%Y-%m-%dT%H:%M:%S%z"))
+    @patch("workspaces.occupeye.archive.MAX_TIME_DELTA", timedelta(days=7))
+    def setUp(self):
+        with open(os.path.join(__location__, "tests_archive_cache.json"), encoding="utf-8") as f:
+            self.archive = OccupEyeArchive(endpoint=TestEndpoint(json.load(f)))
+        self.archive.reset()
+        self.archive.update()
```

Comment:
Maybe a class method to avoid reloading for each test (`setUpTestData` or `setUpClass`).

Reply:
I tried changing to this, I am not sure if I did something wrong but freezegun and patch seem to really dislike setUpClass. The time save is also very minimal showing 0ms for each of the ArchiveViewsTestCases.